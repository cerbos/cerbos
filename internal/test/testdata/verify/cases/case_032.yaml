description: Tests filtered by regexp
config:
  includedTestNamesRegexp: John and his leave request
