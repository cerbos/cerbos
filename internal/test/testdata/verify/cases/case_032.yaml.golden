{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_PASSED",
        "testsCount": 5,
        "resultCounts": [
          {
            "result": "RESULT_SKIPPED",
            "count": 2
          },
          {
            "result": "RESULT_PASSED",
            "count": 3
          }
        ]
      },
      "testCases": [
        {
          "name": "John and his leave request",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "defer",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "With global now",
          "principals": [
            {
              "name": "bev",
              "resources": [
                {
                  "name": "stale_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_SKIPPED",
                        "skipReason": "Test name did not match the provided pattern"
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "With local now",
          "principals": [
            {
              "name": "bev",
              "resources": [
                {
                  "name": "stale_pending_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_SKIPPED",
                        "skipReason": "Test name did not match the provided pattern"
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_PASSED",
    "testsCount": 5,
    "resultCounts": [
      {
        "result": "RESULT_SKIPPED",
        "count": 2
      },
      {
        "result": "RESULT_PASSED",
        "count": 3
      }
    ]
  }
}
